exporter/googlecloudexporter/exporter.go

```diff
@@ -0,0 +1,150 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package googlecloudexporter
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/model/pdata"
+)
+
+// Exporter is a google cloud exporter wrapped with additional functionality
+type Exporter struct {
+	metricsProcessors []component.MetricsProcessor
+	metricsExporter   component.MetricsExporter
+	metricsConsumer   consumer.Metrics
+
+	logsProcessors []component.LogsProcessor
+	logsExporter   component.LogsExporter
+	logsConsumer   consumer.Logs
+
+	tracesProcessors []component.TracesProcessor
+	tracesExporter   component.TracesExporter
+	tracesConsumer   consumer.Traces
+}
+
+// ConsumeMetrics consumes metrics
+func (e *Exporter) ConsumeMetrics(ctx context.Context, md pdata.Metrics) error {
+	if e.metricsConsumer == nil {
+		return nil
+	}
+	return e.metricsConsumer.ConsumeMetrics(ctx, md)
+}
+
+// ConsumeTraces consumes traces
+func (e *Exporter) ConsumeTraces(ctx context.Context, td pdata.Traces) error {
+	if e.tracesConsumer == nil {
+		return nil
+	}
+	return e.tracesConsumer.ConsumeTraces(ctx, td)
+}
+
+// ConsumeLogs consumes logs
+func (e *Exporter) ConsumeLogs(ctx context.Context, ld pdata.Logs) error {
+	if e.logsConsumer == nil {
+		return nil
+	}
+	return e.logsConsumer.ConsumeLogs(ctx, ld)
+}
+
+// Capabilities returns the capabilities of the exporter
+func (e *Exporter) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{MutatesData: true}
```

Comment:
This shouldn't be true, right? In general, exporters should not mutate log records.

Reply:
Actually, you're right, this does mutate records, which has me wondering about how it affects adjacent exporters, or if this is actually OK.

Need to take a closer look.

Reply:
Here's where the warning about doing this is: https://github.com/open-telemetry/opentelemetry-collector/tree/main/exporter#data-ownership

Reply:
Interesting. Do you understand the use case for setting up parallel pipelines with the same receivers? This seems like an extreme edge case.

Reply:
Nevermind. I think I understand the use case after looking closer. Effectively, the issue is having multiple exporters in a single pipeline. There's potential to mutate data that is being sent to the other exporter. Do you see any way around this limitation?

Reply:
It's not that strange to want to send data to two locations at once; for instance, you might want to send to some local prometheus server for short term viewing, but send to GCP maybe for longer term storage (or possibly even another location for REALLY long storage). Another example is debugging using the logging exporter, for example.

Reply:
I think we've had customers ask about this in the past, also, but I'm not sure on the specifics.

Reply:
Only way around the limitation I see is to make a copy of the record before passing to the first consumer.